src/View/Shortcodes/EmbedShortcodeProvider.php

```diff
@@ -141,6 +141,17 @@ protected static function videoEmbed($arguments, $content)
             $arguments['style'] = 'width: ' . intval($arguments['width']) . 'px;';
         }
 
+        // override iframe dimension attributes provided by webservice with ones specified in shortcode arguments
+        foreach (['width', 'height'] as $attr) {
+            if (!($value = $arguments[$attr] ?? false)) {
+                continue;
+            }
+            foreach (['"', "'"] as $quote) {
+                $rx = "/(<iframe .*?)$attr=$quote([0-9]+)$quote([^>]+>)/";
```

Comment:
Do we need to worry about non-pixel values? E.g `100%` or `100vw`?

Reply:
Nope, not with the current implementation.

Non-numeric values in the insert media modal are stripped on page save.  Also there [this line](https://github.com/silverstripe/silverstripe-framework/blob/4.6/src/View/Shortcodes/EmbedShortcodeProvider.php#L160) that forces it to use pixels in the containing div